src/main/java/org/broadinstitute/hellbender/tools/spark/sv/utils/SvCigarUtils.java

```diff
@@ -163,4 +164,80 @@ public static int getUnclippedReadLength(final Cigar cigar) {
                 )
                 .sum();
     }
+
+    /**
+     * Checks the input CIGAR for assumption that operator 'D' is not immediately adjacent to clipping operators.
+     * Then convert the 'I' CigarElement, if it is at either end (terminal) of the input cigar, to a corresponding 'S' operator.
+     * Note that we allow CIGAR of the format '10H10S10I10M', but disallows the format if after the conversion the cigar turns into a giant clip,
+     * e.g. '10H10S10I10S10H' is not allowed (if allowed, it becomes a giant clip of '10H30S10H' which is non-sense).
+     *
+     * @return a pair of number of clipped (hard and soft, including the ones from the converted terminal 'I') bases at the front and back of the
+     *         input {@code cigarAlongInput5to3Direction}.
+     *
+     * @throws IllegalArgumentException when the checks as described above fail.
+     */
+    @VisibleForTesting
+    public static List<CigarElement> checkCigarAndConvertTerminalInsertionToSoftClip(final Cigar cigarAlongInput5to3Direction) {
```

Comment:
It just seemed to me that the fact that the cigar is oriented this way doesn't have any bearing on the actual implementation of this method (unless I'm missing something). So you could just call it `cigar` and have a shorter name without losing any detail. Definitely not a big deal, though.